pkg/tsdb/influxdb/influxql/converter/converter.go

```diff
@@ -0,0 +1,395 @@
+package converter
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/grafana/grafana/pkg/tsdb/influxdb/influxql/util"
+	"github.com/grafana/grafana/pkg/tsdb/influxdb/models"
+	"github.com/grafana/grafana/pkg/util/converter/jsonitere"
+)
+
+func rspErr(e error) *backend.DataResponse {
+	return &backend.DataResponse{Error: e}
+}
+
+func ReadInfluxQLStyleResult(jIter *jsoniter.Iterator, query *models.Query) *backend.DataResponse {
+	iter := jsonitere.NewIterator(jIter)
+	var rsp *backend.DataResponse
+
+	// frameName is pre-allocated. So we can reuse it, saving memory.
+	// It's sized for a reasonably-large name, but will grow if needed.
+	frameName := make([]byte, 0, 128)
+
+l1Fields:
+	for l1Field, err := iter.ReadObject(); ; l1Field, err = iter.ReadObject() {
+		if err != nil {
+			return rspErr(err)
+		}
+		switch l1Field {
+		case "results":
+			rsp = readResults(iter, frameName, query)
+			if rsp.Error != nil {
+				return rsp
+			}
+		case "":
+			if err != nil {
+				return rspErr(err)
+			}
+			break l1Fields
+		default:
+			v, err := iter.Read()
+			if err != nil {
+				rsp.Error = err
+				return rsp
+			}
+			fmt.Println(fmt.Printf("[ROOT] TODO, support key: %s / %v\n\n", l1Field, v))
+		}
+	}
+
+	return rsp
+}
+
+func readResults(iter *jsonitere.Iterator, frameName []byte, query *models.Query) *backend.DataResponse {
+	var rsp *backend.DataResponse
+l1Fields:
+	for more, err := iter.ReadArray(); more; more, err = iter.ReadArray() {
+		if err != nil {
+			return rspErr(err)
+		}
+		for l1Field, err := iter.ReadObject(); l1Field != ""; l1Field, err = iter.ReadObject() {
+			if err != nil {
+				return rspErr(err)
+			}
+			switch l1Field {
+			case "series":
+				rsp = readSeries(iter, frameName, query)
+			case "":
+				break l1Fields
+			default:
+				_, err := iter.Read()
+				if err != nil {
+					return rspErr(err)
+				}
+			}
+		}
+	}
+
+	return rsp
+}
+
+func readSeries(iter *jsonitere.Iterator, frameName []byte, query *models.Query) *backend.DataResponse {
+	var measurement string
+	var tags map[string]string
+	var columns []string
+	var valueFields data.Fields
+	var hasTimeColumn bool
```

Comment:
Just a style suggestion for better readability:

```suggestion
	var (
		measurement   string
		tags          map[string]string
		columns       []string
		valueFields   data.Fields
		hasTimeColumn bool
	)
```